openapi_python_client/parser/properties/union.py

```diff
@@ -92,6 +91,17 @@ def flatten_union_properties(sub_properties: list[PropertyProtocol]) -> list[Pro
             default_or_error.data = data
             return default_or_error, schemas
         prop = evolve(prop, default=default_or_error)
+
+        if data.discriminator:
+            discriminator_or_error = _parse_discriminator(data.discriminator, sub_properties, schemas)
+            if isinstance(discriminator_or_error, PropertyError):
+                return discriminator_or_error, schemas
+            discriminators_list = [discriminator_or_error, *discriminators_list]
+        if discriminators_list:
+            if error := _validate_discriminators(discriminators_list):
+                return error, schemas
+            prop = evolve(prop, discriminators=discriminators_list)
```

Comment:
Using `attrs.evolve` is standard in this codebase; some classes do have mutable fields, but in general mutation is avoided.